### PR TITLE
Optimized the code

### PR DESCRIPTION
**Pull request by pragamj**
Signed-off-by: pragam <pragam.jain@seagate.com>

# Problem Statement
- Optimized the code to fetch hctl status using api, added assert at pre_check.

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [x] New/Affected tests are executed on Latest Build
-  [x] Attach test execution logs
[TEST-N-NODE_TestContDeployment_20220912040524.log](https://github.com/Seagate/cortx-test/files/9547447/TEST-N-NODE_TestContDeployment_20220912040524.log)
[TEST-31384_TestMultipleConfDeploy_20220912051306.log](https://github.com/Seagate/cortx-test/files/9547448/TEST-31384_TestMultipleConfDeploy_20220912051306.log)
-  [x] Collection tested and no collection error int
[collections_log.txt](https://github.com/Seagate/cortx-test/files/9547449/collections_log.txt)
roduced (`pytest --local True --collect-only`)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: tests/prov/test_cont_deployment.py [hunk 1]
```diff
@@ -66,10 +66,12 @@ def setup_class(cls):
...
         cls.data_disk_size = os.getenv("DATA_DISK_SIZE", cls.deploy_cfg["data_disk_size"])
-        cls.meta_disk_size = os.getenv("METADATA_DISK_SIZE", cls.deploy_cfg["metadata_disk_size"])
+        cls.meta_disk_size = os.getenv("METADATA_DISK_SIZE",
+                                       cls.deploy_cfg["metadata_disk_size"])
         cls.iterations = os.getenv("NO_OF_ITERATIONS")
-        cls.s3_instances_per_node = os.getenv("S3_INSTANCE_PER_NODE",
-                                              cls.deploy_cfg["s3_instances_per_node"])
+        cls.s3_instances_per_node = int(os.getenv("S3_INSTANCE_PER_NODE",
+                                        cls.deploy_cfg["s3_instances_per_node"]))
+        cls.log_disk_flag = bool(distutils.util.strtobool(os.getenv("LOG_DEVICE_FLAG")))
```
Comment (priyankaborawake) [inline tests/prov/test_cont_deployment.py:74]: Should the default value for log_disk_flag be false ?

**Review by pranav-sahasrabudhe — APPROVED** [review]
LGTM

File: config/prov/prov_test.yaml [hunk 2]
```diff
@@ -233,16 +234,16 @@ k8s_cortx_deploy:
...
         mem: 1536
-        cpu: '800m'
+        cpu: '500m'
     motr:
       requests:
         mem: '1Gi'
         cpu: '500m'
       limits:
-        mem: 4096
+        mem: 4
```
Comment (priyankaborawake) [inline config/prov/prov_test.yaml:246]: Need to append Gi here ?

File: libs/prov/prov_k8s_cortx_deploy.py [hunk 3]
```diff
@@ -373,14 +376,6 @@ def _operation_on_worker_node():
                 # system disk will be used mount /mnt/fs-local-volume on worker node
                 self.execute_prereq_cortx(node, self.deploy_cfg["k8s_dir"], system_disk)
 
-            thread_list = []
```
Comment (priyankaborawake) [inline libs/prov/prov_k8s_cortx_deploy.py:376]: We dont need to pull cortx images now ?

Comment (pragamj) [reply libs/prov/prov_k8s_cortx_deploy.py:376]: don't need helm charts take care of it

File: config/prov/prov_test.yaml [hunk 4]
```diff
@@ -256,7 +257,7 @@ k8s_cortx_deploy:
           mem: '1Gi'
           cpu: '500m'
         limits:
-          mem: 2000
+          mem: 2
```
[the remark this answers is not in the document]
Comment (pragamj) [reply config/prov/prov_test.yaml:260]: handle in code

**Review by ketanarlulkar — APPROVED** [review]

File: libs/prov/prov_k8s_cortx_deploy.py [hunk 5]
```diff
@@ -892,7 +888,7 @@ def deploy_cortx_k8s_re_job(master_node_list: list, worker_node_list: list,
         return False, output['result']
 
     @staticmethod
-    def get_hctl_status(node_obj, pod_name: str) -> tuple:
+    def get_hctl_status(node_obj) -> tuple:
```
Comment (priyankaborawake) [inline libs/prov/prov_k8s_cortx_deploy.py:891]: Please make sure all occurrences are change for this function call.

**Review by priyankaborawake — APPROVED** [review]

File: libs/prov/prov_k8s_cortx_deploy.py (hunk 5 above)
Comment (pragamj) [reply libs/prov/prov_k8s_cortx_deploy.py:891]: yes it taken care

**Review by sonalk0209 — APPROVED** [review]
Approving for merge!!

**Pull request merged**